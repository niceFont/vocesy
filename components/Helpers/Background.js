const Background = props => {
	return (
		<div
			style={{
				position: "absolute",
				width: "100%",
				minHeight: "100%",
				backgroundColor: "white"
				/* 				background: "url('/static/background2.jpg')",
								backgroundSize: "cover",
								backgroundAttachment: "fixed",
								backgroundPosition: "center", */
			}}
		>
			{props.children}
		</div>
	)
}

export default Background
