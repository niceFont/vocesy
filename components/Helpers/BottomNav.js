import { Navbar, Nav } from "react-bootstrap"
import Link from "next/link"

const BottomNav = () => {
	return (
		<Navbar
			variant="dark"
			bg="dark"
			style={{
				position: "absolute",
				bottom: 0,
				width: "100%",
			}}>
			<Nav>
				<Nav.Item onClick={() => window.location.href = "mailto:peterwilms@hotmail.de"}>
					<Nav.Link as="a">
						Contact
					</Nav.Link>
				</Nav.Item>
				{/* 				<Nav.Item>
					<Link href="/about">
						<Nav.Link as="a">
							About
						</Nav.Link>
					</Link>
				</Nav.Item> */}
				<Nav.Item>
					<Link href="/privacy">
						<Nav.Link as="a">
							Privacy
						</Nav.Link>
					</Link>
				</Nav.Item>
				<Nav.Item>
					<Link href="/tos">
						<Nav.Link as="a">
							Terms of Service
						</Nav.Link>
					</Link>
				</Nav.Item>
				{/* <Nav.Item>
					<Link href="/cookies">
						<Nav.Link as="a">
							Cookies
						</Nav.Link>
					</Link>
				</Nav.Item> */}
			</Nav>
		</Navbar>
	)
}

export default BottomNav
