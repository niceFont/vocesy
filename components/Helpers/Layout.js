import {Nav,
	NavDropdown,
	Navbar,
	NavbarBrand} from "react-bootstrap"
import Link from "next/link"
import { ExtractName } from "../../lib/utils"

export const Layout = props => {
	return (
		<React.Fragment>
			<Navbar
				style={{
					paddingLeft: "4%",
					paddingRight: "4%",
					backgroundColor: "white",
					borderBottom: "1px solid lightgray"
				}}
				fixed="top"
				variant="light">
				<NavbarBrand>
					<Link href="/">
						<img
							style={{
								width: 120, height: 40 
							}}
							src="/static/logo.png"
						/>
					</Link>
				</NavbarBrand>
				<Nav>
					<Link href="/decks/create">
						<Nav.Link as="a">Create</Nav.Link>
					</Link>
					<Nav.Item>
						<Link href="/decks">
							<Nav.Link as="a">My Decks</Nav.Link>
						</Link>
					</Nav.Item>
				</Nav>
				<Navbar.Collapse className="justify-content-end">
					<Nav>
						{!props.user && (
							<Nav.Item>
								<Link href="/login" replace>
									<Nav.Link as="a">log in</Nav.Link>
								</Link>
							</Nav.Item>
						)}
						{props.user && (
							<div style={{
								display: "flex", flexDirection: "row"
							}}>
								<img style={{
									width: 30, height: 30, borderRadius: "100%", margin: "5px 0 5px 0"
								}} src={props.user.picture}></img>
								<NavDropdown
									title={ExtractName(props.user.displayName).split(" ").join("").toLowerCase()}
									id="nav-dropdown">
									<Link
										href={`/user/${encodeURI(ExtractName(props.user.displayName).split(" ").join("").toLowerCase())}`}>
										<NavDropdown.Item as="a">
										Profile
										</NavDropdown.Item>
									</Link>

									<Link href="/logout">
										<NavDropdown.Item
											style={{
												color: "#e84646" 
											}}
											as="a">
										Log Out
										</NavDropdown.Item>
									</Link>
								</NavDropdown>
							</div>
						)}
					</Nav>
				</Navbar.Collapse>
			</Navbar>
			{props.children}

			<Navbar fixed="bottom" bg="light" variant="dark">
				<Nav>

					<Nav.Item>
						<span>ver. 1.01a Still in Development...</span>
					</Nav.Item>
				</Nav>
			</Navbar>
		</React.Fragment>
	)
}
