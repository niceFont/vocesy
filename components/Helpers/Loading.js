import { Spinner, Container, Row } from "react-bootstrap"

export const Loading = props => {
	return (
		<div>
			{!props.fetched && (
				<Container>
					<Row className="justify-content-center">
						<Spinner size="lg" animation="border" role="status">
							<span className="sr-only">Loading...</span>
						</Spinner>
					</Row>
				</Container>
			)}
		</div>
	)
}
