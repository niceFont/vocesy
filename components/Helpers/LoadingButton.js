import {Button, Spinner} from "react-bootstrap"
import React from "react"


export default function LoadingButton({sending,...props}) {
	return (
		<Button {...props}>
			{sending ? <Spinner animation="border"></Spinner> : 
				<React.Fragment>
					{props.children}
				</React.Fragment>
			}
		</Button>
	)
}
