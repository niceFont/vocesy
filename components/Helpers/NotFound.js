
export const NotFound = () => {


	return (
		<h4> 404 | We could not find what you were looking for.</h4>
	)

}