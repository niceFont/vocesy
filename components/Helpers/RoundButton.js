import { Button } from "react-bootstrap"

const RoundButton = props => {
	return (
		<Button
			style={{
				...props.style,
				borderRadius: 25,
				padding: 0,
			}}
			{...props}
		>
			{props.children}
		</Button>
	)
}

export default RoundButton
