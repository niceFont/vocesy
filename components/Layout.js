import {Nav, Navbar,Container, NavbarBrand, Button} from "react-bootstrap"
import Link from "next/link"
import {useAuth0} from "./Auth0wrapper"

export const Layout = (props) => {

	const {isAuthenticated, loginWithRedirect, logout, user} = useAuth0()

	return (
		<div>
			<Navbar fixed="top" bg="dark" variant="dark">
				<NavbarBrand>
					<Link href="/">
						<h3>Vocesy</h3>
					</Link>
				</NavbarBrand>
				<Navbar.Collapse className="justify-content-end">
					<Nav> 
						<Link href="/new">
							<Nav.Link as="a">
								new
							</Nav.Link>
						</Link>

						{!isAuthenticated && (
					
							<Button onClick={() => loginWithRedirect({})}>
						Log in
							</Button>
						)}	
						{isAuthenticated && typeof user !== "undefined"&&
					<Link href={`/user/${encodeURI(user.name)}`}>
						<Nav.Link as="a">
							{user.name.toLowerCase()}
						</Nav.Link>
					</Link>
						}
						{isAuthenticated && <Button variant="outline-danger" onClick={() => logout()}>log out</Button>}
					</Nav>
				</Navbar.Collapse>	
			</Navbar>
			<Container style={{ marginTop: 200 }}>
				{props.children}
			</Container>

			<Navbar fixed="bottom" bg="dark" variant="dark">
			</Navbar>
		</div>

	)
}


