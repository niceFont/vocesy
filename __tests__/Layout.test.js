/*eslint no-undef: off*/
/*eslint camelcase: off*/
import renderer from "react-test-renderer"
import React from "react"
import { Layout } from "../components/Helpers/Layout"
import {mount} from "enzyme"

describe("Layout", () => {
    
	it("renders", () => {
		const tree = renderer.create(<Layout></Layout>)
    
		expect(tree.toJSON()).toMatchSnapshot()
	})

	it("renders with user", () => {
		const component = mount(<Layout user={{
			displayName:"unitTest", email: "unit@test.de" 
		}}></Layout>)
       
		expect(component.containsAllMatchingElements([
			<a>Create</a>,
			<a>My Decks</a>,
			<a>unittest</a>,
			<a>logout</a>
		])).toEqual(true)
	})

})