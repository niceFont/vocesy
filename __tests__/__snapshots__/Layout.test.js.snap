// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders 1`] = `
Array [
  <nav
    className="navbar navbar-expand-md navbar-light fixed-top"
    style={
      Object {
        "backgroundColor": "white",
        "borderBottom": "1px solid lightgray",
        "paddingLeft": "4%",
        "paddingRight": "4%",
      }
    }
  >
    <span
      className="navbar-brand"
    >
      <img
        onClick={[Function]}
        onMouseEnter={[Function]}
        src="/static/logo.png"
        style={
          Object {
            "height": 40,
            "width": 120,
          }
        }
      />
    </span>
    <button
      aria-controls="main-menu"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="main-menu"
    >
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="tab"
            tabIndex={-1}
          >
            Create
          </a>
        </div>
        <div
          className="nav-item"
        >
          <a
            className="nav-link"
            data-rb-event-key={null}
            disabled={false}
            onClick={[Function]}
            onMouseEnter={[Function]}
            role="tab"
            tabIndex={-1}
          >
            My Decks
          </a>
        </div>
      </div>
    </div>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
    >
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="navbar-nav"
          onKeyDown={[Function]}
        >
          <div
            className="nav-item"
          >
            <a
              className="nav-link"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              onMouseEnter={[Function]}
              role="tab"
              tabIndex={-1}
            >
              log in
            </a>
          </div>
          <div
            className="nav-item"
          >
            <a
              className="nav-link"
              data-rb-event-key={null}
              disabled={false}
              onClick={[Function]}
              onMouseEnter={[Function]}
              role="tab"
              tabIndex={-1}
            >
              sign up
            </a>
          </div>
        </div>
      </div>
    </div>
  </nav>,
  <nav
    className="navbar navbar-expand navbar-dark bg-light fixed-bottom"
  >
    <div
      className="navbar-nav"
      onKeyDown={[Function]}
    >
      <div
        className="nav-item"
      >
        <span>
          ver. 1.01a Still in Development...
        </span>
      </div>
    </div>
  </nav>,
]
`;
